artemis-commons/src/main/java/org/apache/activemq/artemis/core/server/NetworkHealthCheck.java

```diff
@@ -157,7 +157,12 @@ public NetworkHealthCheck parseURIList(String addressList) {
 
    @Override
    protected ActiveMQThreadFactory getThreadFactory() {
-      return new ActiveMQThreadFactory("NetworkChecker", "Network-Checker-", false, getThisClassLoader());
```

Comment:
This is just creating a ThreadFactory.. is it really needed?

I would expect a security check around new Threads, but against a new ActiveMQThreadFactory? 

are you sure about it? just checking it's not a typo?

Reply:
I'd like to see a test case on this. I think the issue might be we have some ThreadFactories which are not using the ActiveMQThreadFactory which should imo ensure the privileged access, and actually we just need to fix those up.

Reply:
We run WildFly test suite with a security manager on (which is how this leak was found) BUT I don't have the runs where this happened so I don't know the symptoms :( for a test ensure that this is resolved.

Reply:
I mean we need a test in ActiveMQ Artemis.

Reply:
So we need this converted to a test, e.g. integration test case, with this PR which shows this, given a certain configuration and a way to automate the detection.

Reply:
Well it seems that the increase is stopping at 28 threads, so it is not leaking

Reply:
I don't see it on WildFly with 2.6.4 if I wait sufficiently for the polls to decrease

Reply:
I think the increase stops when the default size is reached so this doesn't look like an issue

Reply:
@ehsavoie can you close this PR then? And the Jira you opened.

Reply:
@ehsavoie please close this then.